client/my-sites/hosting/main.js

```diff
@@ -233,6 +238,7 @@ export const clickActivate = () =>
 export default connect(
 	( state ) => {
 		const siteId = getSelectedSiteId( state );
+		const site = getSelectedSite( state );
```

Comment:
I think that getting the selected site state is making a new request and clearing the state for that site which makes the UI show the Staging card for a few seconds.

We probably need to avoid making that extra request or waiting to be loaded before showing the Staging card

Reply:
I wasn't able to reproduce that issue, but I experienced this behavior:

https://user-images.githubusercontent.com/36432/223862148-65b964db-0d2b-4747-b5b9-4bd58a1b5a07.mp4

```diff
diff --git a/client/my-sites/hosting/staging-site-card/index.js b/client/my-sites/hosting/staging-site-card/index.js
index 0a63c86fb6..49697bce9c 100644
--- a/client/my-sites/hosting/staging-site-card/index.js
+++ b/client/my-sites/hosting/staging-site-card/index.js
@@ -125,7 +125,7 @@ const StagingSiteCard = ( { disabled, siteId, translate } ) => {
 						},
 					} ) }
 				</p>
-				<Button primary href={ `/home/${ stagingSiteSlug }` } disabled={ disabled }>
+				<Button primary href={ `/hosting-config/${ stagingSiteSlug }` } disabled={ disabled }>
 					<span>{ translate( 'Manage staging site' ) }</span>
 				</Button>
 			</>

```

I think what you're describing is likely a problem across this entire page...